File: caesar.go
package main
import (
"fmt"
"log"
"os"
"bufio"
"sync"
)
var dictionary []string
var hof []string
var wg sync.WaitGroup
func main() {
	infile, err := os.Open("wordle.txt")
    if err != nil {
        log.Fatal(err)
    }
	defer infile.Close()
	scanner := bufio.NewScanner(infile)
	for scanner.Scan() {
		dictionary = append(dictionary, scanner.Text())
	}

	for i:=0; i<26; i++ {
		go generator(i)
		wg.Add(1)
	}
	wg.Wait()
}

func generator(i int) {
	defer wg.Done()
	var caesar [26]string
	var alphabet string = "abcdefghijklmnopqrstuvwxyz"
	var most int = 0
	var wordstested int = 0
	var bestword string = ""
	var count int = 0
	var prevwin bool = false
	for j := 0; j < 26; j++ {
		for k := 0; k < 26; k++ {
			if i == j && j == k {
				wordstested += 676
				continue
			}
			for l := 0; l < 26; l++ {
				if j == k && k == l {
					wordstested += 26
					continue
				}
				for m := 0; m < 26; m++ {
					if k == l && l == m {
						wordstested++
						continue
					}
					prevwin = false
					for n := 0; n<26; n++ {
						caesar[n] = ""
						caesar[n] += string(alphabet[(i+n)%26])
						caesar[n] += string(alphabet[(j+n)%26])
						caesar[n] += string(alphabet[(k+n)%26])
						caesar[n] += string(alphabet[(l+n)%26])
						caesar[n] += string(alphabet[(m+n)%26])
						for _, winner := range hof {
							if caesar[n] == winner {
								prevwin = true
								break
							}
						}
						if prevwin == true {
							break
						}
					}
					if prevwin == true {
						wordstested++
						continue
					}
					count = 0
					for _, shift := range caesar {
						if shift == bestword {
							break
						}
						for _, word := range dictionary {
							if word == shift {
								count++
							}
						}
					}
					if count < most {
						wordstested++
						continue
					}
					wordstested++
					most = count
					bestword = caesar[0]
					hof = append(hof, bestword)
					fmt.Printf("\nTHREAD %s\nWords tested: %d\nCurrent best word: %s\nValid shifts: %d", string(alphabet[i]), wordstested, bestword, most)
				}	
			}	
		}
	}
	fmt.Printf("\nBest word is: %s with %d valid rotated words.\n", bestword, most)
}

File: caesar.cpp
//unions help compiler vectorize
#include "immintrin.h"
#include <iostream>
#include <fstream>
#include <vector>
#include <chrono>

using namespace std;
using namespace std::chrono;

struct Word { 
	int8_t ltr1;
	int8_t ltr2;
	int8_t ltr3;
	int8_t ltr4;
	int8_t ltr5;
};
union WordDelta
{
	uint8_t ltr[4];
	uint32_t word;
};
vector<int8_t> ltr1;
vector<WordDelta> dictionary;
int8_t tmpltr2;
int8_t tmpltr3;
int8_t tmpltr4;
int8_t tmpltr5;

int main()
{
	auto start = high_resolution_clock::now();

	string dictIn;
	fstream infile;
	WordDelta dict = {0};
	infile.open("wordle.txt",ios::in);
	while(!infile.eof()) {
		infile >> dictIn;
		
		ltr1.push_back(dictIn[0]);

		tmpltr2 = dictIn[1] - dictIn[0];
		if(tmpltr2 < 0) tmpltr2 += 26;
		dict.ltr[0] = tmpltr2;

		tmpltr3 = dictIn[2] - dictIn[0];
		if(tmpltr3 < 0) tmpltr3 += 26;
		dict.ltr[1] = tmpltr3;

		tmpltr4 = dictIn[3] - dictIn[0];
		if(tmpltr4 < 0) tmpltr4 += 26;
		dict.ltr[2] = tmpltr4;

		tmpltr5 = dictIn[4] - dictIn[0];
		if(tmpltr5 < 0) tmpltr5 += 26;
		dict.ltr[3] = tmpltr5;
		
		dictionary.push_back(dict);

		if(infile.eof())
			break;
	}
	infile.close();

	vector<Word> bestWords = {};
	int count = 0;
	int most = 0;
	const int size = ltr1.size();
	for(int i = 0; i < size; i++) {
		count = 0;
		for(int j = i+1; j < size; j++) {
			if(	dictionary[i].word == dictionary[j].word ) {
				count++;
			}
		}
		if(count > most){
			most = count;
			bestWords.clear();
			bestWords.push_back({ltr1[i],(signed char)dictionary[i].ltr[0],(signed char)dictionary[i].ltr[1],(signed char)dictionary[i].ltr[2],(signed char)dictionary[i].ltr[3]});
		} else if(count == most){
			bestWords.push_back({ltr1[i],(signed char)dictionary[i].ltr[0],(signed char)dictionary[i].ltr[1],(signed char)dictionary[i].ltr[2],(signed char)dictionary[i].ltr[3]});
		}
	}
	
	auto stop = high_resolution_clock::now();
	auto duration = duration_cast<milliseconds>(stop - start);
	cout << duration.count() << "ms" << endl;

	string bestOut;
	for(Word bestWord : bestWords) {
		bestOut = (char)(bestWord.ltr1);

		bestWord.ltr2 += bestWord.ltr1;
		if(bestWord.ltr2 > 90) bestWord.ltr2 -= 26;
		bestOut += (char)bestWord.ltr2;

		bestWord.ltr3 += bestWord.ltr1;
		if(bestWord.ltr3 > 90) bestWord.ltr3 -= 26;
		bestOut += (char)bestWord.ltr3;

		bestWord.ltr4 += bestWord.ltr1;
		if(bestWord.ltr4 > 90) bestWord.ltr4 -= 26;
		bestOut += (char)bestWord.ltr4;

		bestWord.ltr5 += bestWord.ltr1;
		if(bestWord.ltr5 > 90) bestWord.ltr5 -= 26;
		bestOut += (char)bestWord.ltr5;
		
		cout << "Best word is: " << bestOut << " with " << most << " valid rotated words.\n";
	}
	
	return 0;
}